tools/tests/test_nd_freeze

```diff
@@ -0,0 +1,169 @@
+#!/bin/bash
+
+# Temporary scratch directory to save generated files.
+TMP_DIR="tmp_$((1 + RANDOM % 10000))"
```

Comment:
Why not just `mktemp -d`?

I do see that you remove it in `test_teardown`, and I guess we will take the fact that this directory would remain present if tests fail as a feature, but just FYI you wanted to clean it up upon completion regardless of what has happaned, then you could add line below this line (but before running anything else so it doesn't puke before establishing it):
```shell
trap "rm -rf \"$TMP_DIR\"" 0
```

But since it is not cleaned up and might contain relevant info, better to inform user about which TMP_DIR we are using here (add some `echo`).

ATM I cannot even find this TMP_DIR anywhere upon my crashed attempt to run tests